### PR TITLE
add retry for several APIs missing

### PR DESCRIPTION
**Pull request by haakonvt**
### Internal:
Add meta test to check for "POST retry regex"

### Fixed
- Calls to ... are now retried automatically:
    * Functions API: `list`, `retrieve`, `retrieve_multiple`, `activate`
    * FunctionCalls API: `list`, `retrieve`
    * FunctionSchedules API: `list`, `retrieve`
    * ExtractionPipelines API: `retrieve_multiple`
    * ExtractionPipelineRuns API: `list`
    * Transformations API: `list`, `retrieve`, `retrieve_multiple`, `preview`
    * TransformationJobs API: `retrieve`, `retrieve_multiple`
    * TransformationSchedules API: `retrieve`, `retrieve_multiple`
    * Geospatial API:  `list_feature_types`, `retrieve_feature_types`, `retrieve_features`, `list_features`,
      `search_features`, `stream_features`, `aggregate_features`, `get_coordinate_reference_systems`, `get_raster`, `compute`,
    * UserProfilesAPI: `retrieve`, `search`
    * DocumentsAPI: `search`, `list`, `__call__`, `aggregate_count`, `aggregate_cardinality_values`, `aggregate_cardinality_properties`,
      `aggregate_unique_values`, `aggregate_unique_properties`
    * ThreeDRevisionsAPI: `filter_nodes`

## Checklist:
- [x] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by qTipTip — APPROVED** [review]
Functions-related code looks good! 

Thanks! 🚀

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,24 @@ Changes are grouped as follows
...
+    * ExtractionPipelineRuns API: `list`
+    * Transformations API: `list`, `retrieve`, `retrieve_multiple`, `preview`
+    * TransformationJobs API: `retrieve`, `retrieve_multiple`
+    * TransformationSchedules API: `retrieve`, `retrieve_multiple`
+    * Geospatial API:  `list_feature_types`, `retrieve_feature_types`, `retrieve_features`, `list_features`,
+      `search_features`, `stream_features`, `aggregate_features`, `get_coordinate_reference_systems`, `get_raster`, `compute`,
+    * UserProfilesAPI: `retrieve`, `search`
+    * DocumentsAPI: `search`, `list`, `__call__`, `aggregate_count`, `aggregate_cardinality_values`, `aggregate_cardinality_properties`,
+      `aggregate_unique_values`, `aggregate_unique_properties`
+    * ThreeDRevisionsAPI: `filter_nodes`
```
Comment (haakonvt) [inline CHANGELOG.md:36]:
```suggestion
## [7.19.0] - 2024-02-12
### Fixed
- Calls to ... are now retried automatically:
    * Functions API: `list`, `retrieve`, `retrieve_multiple`, `activate`
    * FunctionCalls API: `list`, `retrieve`
    * FunctionSchedules API: `list`, `retrieve`
    * ExtractionPipelines API: `retrieve_multiple`
    * ExtractionPipelineRuns API: `list`
    * Transformations API: `list`, `retrieve`, `retrieve_multiple`, `preview`
    * TransformationJobs API: `retrieve`, `retrieve_multiple`
    * TransformationSchedules API: `retrieve`, `retrieve_multiple`
    * Geospatial API:  `list_feature_types`, `retrieve_feature_types`, `retrieve_features`, `list_features`,
      `search_features`, `stream_features`, `aggregate_features`, `get_coordinate_reference_systems`, `get_raster`, `compute`,
    * UserProfiles API: `retrieve`, `search`
    * Documents API: `search`, `list`, `__call__`, `aggregate_count`, `aggregate_cardinality_values`, `aggregate_cardinality_properties`,
      `aggregate_unique_values`, `aggregate_unique_properties`
    * ThreeDRevisions API: `filter_nodes`
```

**Review by erlendvollset — APPROVED** [review]
oooh - very nice.

File: cognite/client/_api_client.py [hunk 2]
```diff
@@ -79,22 +79,33 @@
...
-            "/raw/dbs/[^/]+/tables/[^/]+/rows",
-            "/raw/dbs/[^/]+/tables/[^/]+/rows/delete",
-            "/context/entitymatching/(byids|list|jobs)",
-            "/sessions/revoke",
-            "/models/.*",
-            "/units/.*",
-            "/annotations/(list|byids|reverselookup)",
-        )
+        [
+            r"|".join(
```
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/_api_client.py:85]: Speed! 🤩 (To avoid the overhead of a Python loop)

**Pull request merged**